{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from os.path import join\n",
    "import sys\n",
    "import os\n",
    "sys.path.append(\"/home/sunzehui/GeneNet/\")\n",
    "from config_path import *\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "processed_dir = 'processed'\n",
    "data_dir = 'raw_data'\n",
    "\n",
    "processed_dir = join(DATA_PATH, processed_dir)\n",
    "data_dir = join(DATA_PATH, data_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/sunzehui/GeneNet/_database/processed\n",
      "/home/sunzehui/GeneNet/_database/raw_data\n"
     ]
    }
   ],
   "source": [
    "print(processed_dir)\n",
    "print(data_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "preparing mutations ...\n",
      "mutation distribution\n",
      "Missense_Mutation           51002\n",
      "Silent                      21346\n",
      "Intron                      11385\n",
      "Nonsense_Mutation            2830\n",
      "Frame_Shift_Del              2755\n",
      "Splice_Site                  2695\n",
      "3'UTR                        1200\n",
      "Frame_Shift_Ins               915\n",
      "In_Frame_Del                  585\n",
      "5'UTR                         445\n",
      "In_Frame_Ins                   79\n",
      "RNA                            42\n",
      "Nonstop_Mutation               40\n",
      "Stop_Codon_Del                 13\n",
      "Start_Codon_Del                11\n",
      "De_novo_Start_OutOfFrame        5\n",
      "De_novo_Start_InFrame           2\n",
      "Start_Codon_Ins                 2\n",
      "Stop_Codon_Ins                  1\n",
      "lincRNA                         1\n",
      "Name: Variant_Classification, dtype: int64\n",
      "number of mutations 60935.0 60.27200791295747\n",
      "preparing copy number variants ...\n",
      "preparing response ...\n",
      "preparing copy number burden ...\n",
      "Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/sunzehui/.conda/envs/tf/lib/python3.9/site-packages/openpyxl/worksheet/_read_only.py:79: UserWarning: Unknown extension is not supported and will be removed\n",
      "  for idx, row in parser.parse():\n",
      "/home/sunzehui/.conda/envs/tf/lib/python3.9/site-packages/openpyxl/worksheet/_read_only.py:79: UserWarning: Unknown extension is not supported and will be removed\n",
      "  for idx, row in parser.parse():\n"
     ]
    }
   ],
   "source": [
    "def prepare_design_matrix_crosstable():\n",
    "    print('preparing mutations ...')\n",
    "\n",
    "    filename = '41588_2018_78_MOESM4_ESM.txt'\n",
    "    id_col = 'Tumor_Sample_Barcode'\n",
    "    df = pd.read_csv(join(data_dir, filename), sep='\\t', low_memory=False, skiprows=1)\n",
    "    print ('mutation distribution')\n",
    "    print (df['Variant_Classification'].value_counts())\n",
    "\n",
    "    if filter_silent_muts:\n",
    "        df = df[df['Variant_Classification'] != 'Silent'].copy()\n",
    "    if filter_missense_muts:\n",
    "        df = df[df['Variant_Classification'] != 'Missense_Mutation'].copy()\n",
    "    if filter_introns_muts:\n",
    "        df = df[df['Variant_Classification'] != 'Intron'].copy()\n",
    "\n",
    "    # important_only = ['Missense_Mutation', 'Nonsense_Mutation', 'Frame_Shift_Del', 'Splice_Site','Frame_Shift_Ins', 'In_Frame_Del', 'In_Frame_Ins', 'Start_Codon_SNP','Nonstop_Mutation', 'De_novo_Start_OutOfFrame', 'De_novo_Start_InFrame']\n",
    "    exclude = ['Silent', 'Intron', \"3\\'UTR\", \"5\\'UTR\", 'RNA', 'lincRNA']\n",
    "    if keep_important_only:\n",
    "        df = df[~df['Variant_Classification'].isin(exclude)].copy()\n",
    "    if truncating_only:\n",
    "        include = ['Nonsense_Mutation', 'Frame_Shift_Del', 'Frame_Shift_Ins']\n",
    "        df = df[df['Variant_Classification'].isin(include)].copy()\n",
    "    df_table = pd.pivot_table(data=df, index=id_col, columns='Hugo_Symbol', values='Variant_Classification',\n",
    "                              aggfunc='count')\n",
    "    df_table = df_table.fillna(0)\n",
    "    total_numb_mutations = df_table.sum().sum()\n",
    "\n",
    "    number_samples = df_table.shape[0]\n",
    "    print('number of mutations', total_numb_mutations, total_numb_mutations / (number_samples + 0.0))\n",
    "    filename = join(processed_dir, 'somatic_mutations' + ext + '.csv')\n",
    "    df_table.to_csv(filename)\n",
    "\n",
    "\n",
    "def prepare_response():\n",
    "    print('preparing response ...')\n",
    "    filename = '41588_2018_78_MOESM5_ESM.xlsx'\n",
    "    df = pd.read_excel(join(data_dir, filename), sheet_name='Supplementary_Table3.txt', skiprows=2)\n",
    "    response = pd.DataFrame()\n",
    "    response['id'] = df['Patient.ID']\n",
    "    response['response'] = df['Sample.Type']\n",
    "    response['response'] = response['response'].replace('Metastasis', 1)\n",
    "    response['response'] = response['response'].replace('Primary', 0)\n",
    "    response = response.drop_duplicates()\n",
    "    response.to_csv(join(processed_dir, 'response.csv'), index=False)\n",
    "\n",
    "\n",
    "def prepare_cnv():\n",
    "    print('preparing copy number variants ...')\n",
    "    filename = '41588_2018_78_MOESM10_ESM.txt'\n",
    "    df = pd.read_csv(join(data_dir, filename), sep='\\t', low_memory=False, skiprows=1, index_col=0)\n",
    "    df = df.T\n",
    "    df = df.fillna(0.)\n",
    "    filename = join(processed_dir, 'CNV.csv')\n",
    "    df.to_csv(filename)\n",
    "\n",
    "\n",
    "def prepare_cnv_burden():\n",
    "    print('preparing copy number burden ...')\n",
    "    filename = '41588_2018_78_MOESM5_ESM.xlsx'\n",
    "    df = pd.read_excel(join(data_dir, filename), skiprows=2, index_col=1)\n",
    "    cnv = df['Fraction of genome altered']\n",
    "    filename = join(processed_dir, 'CNV_burden.csv')\n",
    "    cnv.to_frame().to_csv(filename)\n",
    "\n",
    "\n",
    "# remove silent and intron mutations\n",
    "filter_silent_muts = False\n",
    "filter_missense_muts = False\n",
    "filter_introns_muts = False\n",
    "keep_important_only = True\n",
    "truncating_only = False\n",
    "\n",
    "ext = \"\"\n",
    "if keep_important_only:\n",
    "    ext = '_important_only'\n",
    "\n",
    "if truncating_only:\n",
    "    ext = 'truncating_only'\n",
    "\n",
    "if filter_silent_muts:\n",
    "    ext = \"_no_silent\"\n",
    "\n",
    "if filter_missense_muts:\n",
    "    ext = ext + \"_no_missense\"\n",
    "\n",
    "if filter_introns_muts:\n",
    "    ext = ext + \"_no_introns\"\n",
    "\n",
    "prepare_design_matrix_crosstable()\n",
    "prepare_cnv()\n",
    "prepare_response()\n",
    "prepare_cnv_burden()\n",
    "print('Done')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check the processed data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/sunzehui/GeneNet/_database'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DATA_PATH"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`CNV.csv` relects how each tumer sample barcode's gene copy number variation situation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>PIK3CD</th>\n",
       "      <th>MTOR</th>\n",
       "      <th>JUN</th>\n",
       "      <th>NRAS</th>\n",
       "      <th>NOTCH2</th>\n",
       "      <th>RIT1</th>\n",
       "      <th>NTRK1</th>\n",
       "      <th>DDR2</th>\n",
       "      <th>MDM4</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAM</th>\n",
       "      <th>RNF26</th>\n",
       "      <th>C1QTNF5</th>\n",
       "      <th>MFRP</th>\n",
       "      <th>USP2</th>\n",
       "      <th>LOC100499227</th>\n",
       "      <th>THY1</th>\n",
       "      <th>PVRL1</th>\n",
       "      <th>TRIM29</th>\n",
       "      <th>OAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00-029N9_LN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01-087MM_BONE</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>01-095N1_LN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>01-120A1_LIVER</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-2.0</td>\n",
       "      <td>-2.0</td>\n",
       "      <td>-2.0</td>\n",
       "      <td>-2.0</td>\n",
       "      <td>-2.0</td>\n",
       "      <td>-2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>02-083E1_LN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1008</th>\n",
       "      <td>TP_2069</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1009</th>\n",
       "      <td>TP_2077</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1010</th>\n",
       "      <td>TP_2078</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1011</th>\n",
       "      <td>TP_2079</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1012</th>\n",
       "      <td>PROS11496-6115321-SM-6CNQ5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1013 rows Ã— 13803 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Unnamed: 0  PIK3CD  MTOR  JUN  NRAS  NOTCH2  RIT1  \\\n",
       "0                    00-029N9_LN     0.0   0.0  0.0   0.0     0.0   0.0   \n",
       "1                  01-087MM_BONE     0.0   0.0  0.0   0.0     0.0   0.0   \n",
       "2                    01-095N1_LN     1.0   0.0  0.0   0.0     0.0   0.0   \n",
       "3                 01-120A1_LIVER     0.0   0.0  0.0   0.0     0.0   0.0   \n",
       "4                    02-083E1_LN     1.0   0.0  0.0   0.0     0.0   0.0   \n",
       "...                          ...     ...   ...  ...   ...     ...   ...   \n",
       "1008                     TP_2069     0.0   0.0  0.0   0.0     0.0   0.0   \n",
       "1009                     TP_2077     0.0   0.0  0.0   0.0     0.0   0.0   \n",
       "1010                     TP_2078     0.0   0.0  0.0   0.0     0.0   0.0   \n",
       "1011                     TP_2079     0.0   0.0  0.0   0.0     0.0   0.0   \n",
       "1012  PROS11496-6115321-SM-6CNQ5     0.0   0.0  0.0   0.0     0.0   0.0   \n",
       "\n",
       "      NTRK1  DDR2  MDM4  ...  MCAM  RNF26  C1QTNF5  MFRP  USP2  LOC100499227  \\\n",
       "0       0.0   0.0   0.0  ...   0.0    0.0      0.0   0.0   0.0           0.0   \n",
       "1       0.0   0.0   0.0  ...  -1.0   -1.0     -1.0  -1.0  -1.0          -1.0   \n",
       "2       1.0   0.0   0.0  ...   1.0    1.0      1.0   1.0   1.0           1.0   \n",
       "3       0.0   0.0   0.0  ...   0.0    0.0      0.0   0.0  -2.0          -2.0   \n",
       "4       0.0   0.0   1.0  ...   2.0    2.0      2.0   2.0   2.0           2.0   \n",
       "...     ...   ...   ...  ...   ...    ...      ...   ...   ...           ...   \n",
       "1008    0.0   0.0   0.0  ...   0.0    0.0      0.0   0.0   0.0           0.0   \n",
       "1009    0.0   0.0   0.0  ...   0.0    0.0      0.0   0.0   0.0           0.0   \n",
       "1010    0.0   0.0   1.0  ...   0.0    0.0      0.0   0.0   0.0           0.0   \n",
       "1011    0.0   0.0   0.0  ...   0.0    0.0      0.0   0.0   0.0           0.0   \n",
       "1012    0.0   0.0   0.0  ...  -1.0   -1.0     -1.0  -1.0  -1.0          -1.0   \n",
       "\n",
       "      THY1  PVRL1  TRIM29  OAF  \n",
       "0      0.0    0.0     0.0  0.0  \n",
       "1     -1.0   -1.0    -1.0 -1.0  \n",
       "2      1.0    1.0     1.0  1.0  \n",
       "3     -2.0   -2.0    -2.0 -2.0  \n",
       "4      2.0    2.0     2.0  2.0  \n",
       "...    ...    ...     ...  ...  \n",
       "1008   0.0    0.0     0.0  0.0  \n",
       "1009   0.0   -1.0    -1.0  0.0  \n",
       "1010   0.0    0.0     0.0  0.0  \n",
       "1011   0.0    0.0     0.0  0.0  \n",
       "1012  -1.0   -1.0    -1.0  0.0  \n",
       "\n",
       "[1013 rows x 13803 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv = pd.read_csv(os.path.join(DATA_PATH, \"processed/CNV.csv\") ) \n",
    "cnv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`cnv_burden.csv` reflects how each tumor sample barcode's burden situation, it's calculated from cnvs themselves."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient.ID</th>\n",
       "      <th>Fraction of genome altered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAPC-STID0000011640-Tumor-SM-2XU1H</td>\n",
       "      <td>0.010487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAPC-STID0000021561-Tumor-SM-3RVWB</td>\n",
       "      <td>0.135831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAPC-STID0000011949-Tumor-SM-2XU1I</td>\n",
       "      <td>0.190097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAPC-STID0000021610-Tumor-SM-2XU13</td>\n",
       "      <td>0.054238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAPC-STID0000021537-Tumor-SM-3RVW7</td>\n",
       "      <td>0.054551</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Patient.ID  Fraction of genome altered\n",
       "0  AAPC-STID0000011640-Tumor-SM-2XU1H                    0.010487\n",
       "1  AAPC-STID0000021561-Tumor-SM-3RVWB                    0.135831\n",
       "2  AAPC-STID0000011949-Tumor-SM-2XU1I                    0.190097\n",
       "3  AAPC-STID0000021610-Tumor-SM-2XU13                    0.054238\n",
       "4  AAPC-STID0000021537-Tumor-SM-3RVW7                    0.054551"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_burden = pd.read_csv(os.path.join(DATA_PATH, 'processed/CNV_burden.csv'))\n",
    "cnv_burden.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.13 ('tf')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "vscode": {
   "interpreter": {
    "hash": "4d41b3e66ab4053a365be12b1f0ec52700f833eebdee548abd3d6dad6adc98e8"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
